src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -47,6 +56,11 @@ private bool IsValidScanTarget(string file)
 
         protected override BinaryAnalyzerContext CreateContext(AnalyzeOptions options, IAnalysisLogger logger, RuntimeConditions runtimeErrors, PropertiesDictionary policy = null, string filePath = null)
         {
+            if (logger is AggregatingLogger aggregatingLogger && this.Telemetry?.TelemetryClient != null)
```

Comment:
>AggregatingLogger [](http://example.com/codeflow?start=26&length=17)

So, if the input logger is _not_ an aggregating logger, should we create one, add the original sole logger to it and then add the telemetry logger?

Reply:
Interesting. I tried something like that originally by creating a custom logger that wraps the passed-in logger, and passing the custom logger to base.CreateContext. It failed in CreateContext because there's an unconditional cast to AggregatingLogger. However, if I use another AggregatingLogger, that should work.

Reply:
I looked into this and decided to leave it the way it is for now. The problem with handling the "not AggregatingLogger" case is that this method (CreateContext) gets called again for each file with a CachedLogger and, at least for now, we don't want to add a RuleTelemetryLogger to those.

I think CreateContext is the wrong place to add this code, but it's the only customization point we have at the moment.

I think we should have a virtual "CreateLogger" or "ConfigureLogger" method that is called just once from the base class -- at the point where we currently create the AggregatingLogger. But that requires a Sarif SDK change.